### PR TITLE
Insert new feature: custom dns records

### PR DESCRIPTION
**Pull request by Lukas0610**
This patch would add the possibility to add custom dns entries to each domain.
The custom entries are appended at the bottom of the zonefile generated by Froxlor.
Users can add custom entries at a text field while editing/adding domains.

### CODE REVIEW

File: lng/dutch.lng.php [hunk 1]
```diff
@@ -284,6 +284,7 @@
 $lng['admin']['templates']['USERNAME'] = 'Vervangen door de gebruikersnaam van de klant.';
 $lng['admin']['templates']['PASSWORD'] = 'Vervangen door het wachtwoord van de klant.';
 $lng['admin']['templates']['EMAIL'] = 'Vervangen door het adres van het POP3/IMAP account.';
+$lng['admin']['dnsrecords'] = 'Own DNS-Records which should be saved in the zonefile<br /><strong>Attention:</strong> The code won\'t be checked for any errors. If it contains errors, webserver might not start again!'; // ToDo: Translante
```
Comment (d00p) [inline lng/dutch.lng.php]: Why should the webserver not start again if there are issues in the bind config?!

File: lng/french.lng.php [hunk 2]
```diff
@@ -432,6 +432,7 @@
 $lng['panel']['decending'] = 'descendant';
 $lng['panel']['search'] = 'Rechercher';
 $lng['panel']['used'] = 'utilisé';
+$lng['admin']['dnsrecords'] = 'Own DNS-Records which should be saved in the zonefile<br /><strong>Attention:</strong> The code won\'t be checked for any errors. If it contains errors, webserver might not start again!'; // ToDo: Translante
```
Comment (d00p) [inline lng/french.lng.php]: Do not add english language strings to another language file. English is the fallback if there is no string in the correspoding language-file.

File: lng/italian.lng.php [hunk 3]
```diff
@@ -309,6 +309,7 @@
 $lng['admin']['templates']['PASSWORD'] = 'Rimpiazzato con la password dell\'account.';
 $lng['admin']['templates']['EMAIL'] = 'Rimapiazzato con l\'indirizzo dell\'account.';
 $lng['admin']['webserver'] = 'Webserver';
+$lng['admin']['dnsrecords'] = 'Own DNS-Records which should be saved in the zonefile<br /><strong>Attention:</strong> The code won\'t be checked for any errors. If it contains errors, webserver might not start again!'; // ToDo: Translante
```
Comment (d00p) [inline lng/italian.lng.php]: Do not add english language strings to another language file. English is the fallback if there is no string in the correspoding language-file.

File: lng/portugues.lng.php [hunk 4]
```diff
@@ -312,6 +312,7 @@
 $lng['admin']['templates']['USERNAME'] = 'Altere para nome da conta do cliente.';
 $lng['admin']['templates']['PASSWORD'] = 'Altere com a senha da conta do cliente.';
 $lng['admin']['templates']['EMAIL'] = 'Altere com os dados do servidor POP3/IMAP.';
+$lng['admin']['dnsrecords'] = 'Own DNS-Records which should be saved in the zonefile<br /><strong>Attention:</strong> The code won\'t be checked for any errors. If it contains errors, webserver might not start again!'; // ToDo: Translante
```
Comment (d00p) [inline lng/portugues.lng.php]: Do not add english language strings to another language file. English is the fallback if there is no string in the correspoding language-file.

File: lng/swedish.lng.php [hunk 5]
```diff
@@ -301,6 +301,7 @@
 $lng['admin']['templates']['USERNAME'] = 'Ändra till kundens kontonamns användarnamn.';
 $lng['admin']['templates']['PASSWORD'] = 'Ändra till kundens kontonamns lösenord.';
 $lng['admin']['templates']['EMAIL'] = 'Ändra till adressen för POP3/IMAP kontot.';
+$lng['admin']['dnsrecords'] = 'Own DNS-Records which should be saved in the zonefile<br /><strong>Attention:</strong> The code won\'t be checked for any errors. If it contains errors, webserver might not start again!'; // ToDo: Translante
```
Comment (d00p) [inline lng/swedish.lng.php]: Do not add english language strings to another language file. English is the fallback if there is no string in the correspoding language-file.

File: lng/dutch.lng.php (hunk 1 above)
Comment (Lukas0610) [inline lng/dutch.lng.php]: Okay, the first one was a mistake, the second one I didn't know

File: scripts/jobs/cron_tasks.inc.dns.10.bind.php [hunk 6]
```diff
@@ -365,6 +365,17 @@ protected function generateZone($domain, $froxlorhost = false) {
...
 		}
+		
+		// Append custom DNS-Records
+		$domain_stmt = Database::prepare("
+			SELECT `dnsrecords` FROM `" . TABLE_PANEL_DOMAINS . "`
+			WHERE `id` = :domainid
+		");
+		Database::pexecute($domain_stmt, array('domainid' => $domain['id']));
+						
+		while ($dnsrecords = $domain_stmt->fetch(PDO::FETCH_ASSOC)) {
```
Comment (d00p) [inline scripts/jobs/cron_tasks.inc.dns.10.bind.php]: Why re-read the dnsrecords field, superfluous query with even more superfluous while-loop? Just add the dnsrecords field to the query in line 80 of this file and use 
```php
$zonefile .= $domain['dnsrecords'];
```

File: install/updates/froxlor/0.9/update_0.9.inc.php [hunk 7]
```diff
@@ -3002,5 +3002,7 @@
     showUpdateStep("Updating from 0.9.34-dev4 to 0.9.34 final");
     lastStepStatus(0);
 
+	Database::query("ALTER TABLE `".TABLE_PANEL_DOMAINS."` ADD `dnsrecords` text;");
```
Comment (d00p) [inline install/updates/froxlor/0.9/update_0.9.inc.php]: Wrong place for the database update, people using 0.9.34 final or later will NEVER get this update, always append a new -devX version when db updates are needed (e.g 0.9.35-dev1)

**Comment by d00p** [comment]
as of https://github.com/Froxlor/Froxlor/tree/dns-editor, froxlor will soon be having a full qualified DNS editor so I think this PR is unnecessary

**Pull request closed without merging**